libbeat/processors/actions/rename_fields.go

```diff
@@ -0,0 +1,103 @@
+package actions
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type renameFields struct {
+	Fields        []fromTo
+	ignoreMissing bool
+	failOnError   bool
+}
+
+type fromTo struct {
+	From string `config:"from"`
+	To   string `config:"to"`
+}
+
+func init() {
+	processors.RegisterPlugin("rename_fields",
+		configChecked(newRenameFields,
+			requireFields("fields")))
+}
+
+func newRenameFields(c *common.Config) (processors.Processor, error) {
+	config := struct {
```

Comment:
There's some duplication in this anonymous struct and the `renameFields` struct. Perhaps you could elevate this anonymous part to `type renameFieldsConfig struct` and then embed it in the `renameFields` processor.

```
type renameFields struct {
    renameFieldsConfig
}

type renameFieldsConfig struct {
    Fields        []fromTo `config:"fields"`
    IgnoreMissing bool     `config:"ignore_missing"`
    FailOnError   bool     `config:"fail_on_error"`
}
```

Reply:
done